sklearn/gaussian_process/_gpr.py

```diff
@@ -358,18 +355,12 @@ def predict(self, X, return_std=False, return_cov=False):
 
                 return y_mean, y_cov
             elif return_std:
-                # cache result of K_inv computation
-                if self._K_inv is None:
-                    # compute inverse K_inv of K based on its Cholesky
-                    # decomposition L and its inverse L_inv
-                    L_inv = solve_triangular(self.L_.T,
-                                             np.eye(self.L_.shape[0]))
-                    self._K_inv = L_inv.dot(L_inv.T)
+                v = cho_solve((self.L_, True), K_trans.T)  # Line 5
```

Comment:
Maybe an inline comment with the math formula in pseudo latex or ASCII notation would help understanding what's happening in this branch when `return_std is True`

Reply:
Also, using a lower case variable name implies that v is a vector but as I understand, it has shape `(n_samples_train, n_samples_pred)` if I am not mistaken.

I think we should use an upper case variable name here to make the code less confusing. I am not sure how to names this.

Reply:
The variable `v` should also be renamed in the `return_cov` case for consistency.

Reply:
Regarding the formula, I added a PR with the equation and the references to be more explicit there: https://github.com/scikit-learn/scikit-learn/pull/19952